src/parameterFiles.ts

```diff
@@ -0,0 +1,507 @@
+// ----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// ----------------------------------------------------------------------------
+
+import * as assert from 'assert';
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { commands, ConfigurationTarget, MessageItem, TextDocument, Uri, ViewColumn, window, workspace } from 'vscode';
+import { callWithTelemetryAndErrorHandling, DialogResponses, IActionContext, IAzureQuickPickItem, UserCancelledError } from 'vscode-azureextensionui';
+import { configKeys, configPrefix, globalStateKeys, isWin32 } from './constants';
+import { DeploymentTemplate } from './DeploymentTemplate';
+import { ext } from './extensionVariables';
+import { containsParamsSchema } from './schemas';
+
+const readAtMostBytesToFindParamsSchema = 4 * 1024;
+const currentMessage = "Current";
+const similarFilenameMessage = "Similar filename";
+const howToMessage = `You can manually associate a parameter file with this template at any time by clicking "Select Parameter File..." in the status bar or the editor context menu.`;
+
+const _filesToIgnoreThisSession: Set<string> = new Set<string>();
+
+interface IPossibleParamFile {
+  uri: Uri;
+  friendlyPath: string;
+  isCloseNameMatch: boolean;
+}
+
+// tslint:disable-next-line: max-func-body-length
+export async function selectParameterFile(actionContext: IActionContext, sourceUri: Uri | undefined): Promise<void> {
+  if (!sourceUri) {
+    sourceUri = window.activeTextEditor?.document.uri;
+  }
+  if (!sourceUri) {
+    await ext.ui.showWarningMessage(`No Azure Resource Manager template file is being edited.`);
+    return;
+  }
+
+  let templateUri: Uri = sourceUri;
+
+  // Verify it's a template file
+  const contents = (await fse.readFile(templateUri.fsPath, { encoding: "utf8" })).toString();
+  const template: DeploymentTemplate = new DeploymentTemplate(contents, "Check file is template");
```

Comment:
>DeploymentTemplate [](start = 43, length = 18)

If we are using activeTextEditor, would there already be existing DeploymentTemplate so we don't have to check this again?